Start adding localization

a2937 opened the pull request:
Excuse me, but would you test to see if what I have localized so far works.

a2937 committed:
Start adding localization

a2937 commented:
If it does work, I will finish the localisations.

JanVargovsky commented:
Thanks for the contribution Aaron! Unfortunately, I have very limited time because I'm finishing my master thesis this month. Could you try to setup your own bot on own server and test it? I've added a quick guide how to do it in the readme.

Overall, your proposed solution has a lot defects, namely unused variables, wrong formatting etc. but more importantly I doubt it is able to handle multi-language load. The bot may listen on multiple servers that have to be independent - from the language perspective.

Side note: The Discord.NET library that I depend on released its 2nd version a few months ago. I still did not upgrade thought. I would do some research how does other bots handle localisation, maybe they did add some native support or somebody created an extension for it.

a2937 committed:
Have a proper functional localization

I didn't actually need a localization service. It was built in code generation automatically. The strings can be retrieved from the resources file automatically and are dependent on the supplied thread.

a2937 commented:
I managed to get it to work on my server properly. I fixed things up. I decided against a localization service because it was unnecessary. That functionality was built into C# projects by default. Updating to the newest version of Discord.NET is going to be a pain, if trying to update the clean module and then reverting back to a prior version were an indicator. There's more places where the summary needs to be changed to a key name however.

a2937 committed:
Translate some more phrases

Still have more to go.

JanVargovsky commented:
I really appreciate your effort but can you revert cosmetic changes, please? It is really painful to do a code review that literally changes all files. For instance adding/removing private access modifier, change from explicit type to var vice versa, adding/removing brackets, adding/removing white spaces, newline at the end of file, converting standard methods to arrow functions etc.

a2937 commented:
I will do my best to try to undo my cosmetic changes.

a2937 committed:
Revert a major chunk of the cosmetic style changes I made earlier.

a2937 committed:
Revert some more cosmetic changes.

a2937 committed:
Revert more cosmetic changes.

a2937 committed:
Changes in both raid module and raid boss info service were applied.

a2937 committed:
More slight cosmetic reversions.

a2937 committed:
Revert spacing as much as I can.

a2937 committed:
Use the  DEBUG constant defined by Visual Studio if the environment variable is not found. Also try to create a nice text file for letting users know to contribute more languages.
